src/test/integration/integration-test.py

```diff
@@ -0,0 +1,92 @@
+            )
+            utils.check_schema(self, response, spending_schema)
+            response_data = response.json()['data']
+            actual_onid = response_data['id']
+            self.assertEqual(actual_onid.lower(), onid.lower())
+
+    # Test case: GET /spendingauthority with invalid authority onids
+    def test_get_non_authority_onids(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['invalid_authority_onids']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = []
+            self.assertEqual(response.json()['data'], spending_schema)
+
+    # Test case: GET /spendingauthority with bad request
+    def test_error_response(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['bad_request']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 400, params=params)
+            error_schema = utils.get_resource_schema(self, 'Error')
+            utils.check_schema(self, response, error_schema)
+
+    # Test case: GET /spendingauthority spending limits
+    def test_spending_limits(self, endpoint='/spendingauthority'):
+        # Checking that the lists of spending limits and indexes are unique
+        # Checking valid onids with spending authority only
+        for onid in self.test_cases['valid_authority_onids']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            response_data = response.json()['data']
+            attributes = response_data['attributes']
+            limits = attributes['limits']
+            spendingLimit = []
+            index_array = []
+            # Checking that each spending limit has at least one index
+            for limit in limits:
+                for index in limit['indexes']:
+                    index_array.append(index)
+                spendingLimit.append(limit['spendingLimit'])
+                self.assertGreater(len(limit['indexes']), 0)
+            # Creating a dict indexes_list from indexes,
+            # and compare it with the original array of indexes
+            indexes_list = list(dict.fromkeys(index_array))
+            self.assertEqual(index_array, indexes_list)
+            # Creating a dict spendingLimit_list from spendingLimit,
+            # and compare it with the original array of spendingLimit
+            spendingLimit_list = list(dict.fromkeys(spendingLimit))
+            self.assertEqual(spendingLimit, spendingLimit_list)
+
+
+if __name__ == '__main__':
+    arguments, argv = utils.parse_arguments()
+
+    # Setup logging level
+    if arguments.debug:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+
+    integration_tests.setup(arguments.config_path, arguments.openapi_path)
+    unittest.main(argv=argv, exit=False)
```

Comment:
This is a known bug which @wilsonia found recently. `exit=False` should be removed so that Jenkins job will fail correctly if the integration test doesn't pass. @wilsonia also have another ticket (CO-1358) to update the code in skeleton repo.